### ISSUE TITLE
Feedback: iA3 is amazing

### ISSUE THREAD

**Issue by bmaltais** [KohakuBlueleaf/LyCORIS#63]
I know you mention that with iA3 "_You can only get reasonable result on the model you trained on_"... but it does! Actually I get better results with the iA3 on the model I did not train with. Here is a sample I trained of Buster Keaton using SD1.5 Base and rendered using Realistic Vision 3.0:

![05343-1750692388-color photo of BstrKtn posing on a balcony, photorealistic, film grain, wearing a blue shirt, film grain, bokeh_](https://github.com/KohakuBlueleaf/LyCORIS/assets/7474674/a7d6acc8-bc4d-4571-870d-42d4ba8d8817)

I mean... for a 208KB file it is bang on accurate. This make me wonder how much waste data is in the other LoRA type.

I trained using the Prodigy optimizer and one thing that was noticed by #Dogucat is that using batch of 1 and gradient 1 on roughly 30 images will result in a very nice prodigy dlr curve... and this is what you want to get a nice looking model... in theory:

![image](https://github.com/KohakuBlueleaf/LyCORIS/assets/7474674/b5ddbcbb-4b19-4027-b453-1638dc67707f)

Here is the model... so small! 167KB once zipped... could probably just be embedded in the image metadata.

[Buster Keaton-3.0b-ia3-sd15.zip](https://github.com/KohakuBlueleaf/LyCORIS/files/11900417/Buster.Keaton-3.0b-ia3-sd15.zip)

Trigger work is: BstrKtn

**Comment by bmaltais** [comment]
Tried training a version with two concepts (with a hat and without) and it worked great.

![05367-4019062888-color photo of buster keaton hat posing on a balcony, photorealistic, film grain, bokeh_](https://github.com/KohakuBlueleaf/LyCORIS/assets/7474674/5d34ef6d-451c-4411-9cd4-950a0106e197)

**Comment by 2blackbar** [comment]
Can we use it in the gui with no issues? Is it in Your dev2 or we have to tinker with text and add it?

**Comment by bmaltais** [comment]
It is the dev2 branch of the gui that support the requirec parameters for Lycoris ia3. It is resdy to produce models that gave the results above.

**Comment by 2blackbar** [comment]
Can You share Your configuration file for the gui  so i can open it ? I dont have good results, maybe cosine is the issue

**Comment by KohakuBlueleaf** [comment]
you may also try the lokr
which has quite similar property with IA3

**Comment by bmaltais** [comment]
> Can You share Your configuration file for the gui so i can open it ? I dont have good results, maybe cosine is the issue but i changed it to constant and my previews look nothing like my subject

You can grab it from here: https://www.reddit.com/r/StableDiffusion/comments/14low8y/lora_lycoris_ia3_is_amazing_info_in_1st_comment/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=2&utm_content=share_button

**Comment by 2blackbar** [comment]
I used your settings but learning rates are so high  ( 1 instead of usual 0000.1) i get just distorted rectangles as outputs
I think webui should have own default settings from get go so we dont have to guess what works and theres no centralised place to store all the settings for all types of locon lycoris loha variants

**Comment by bmaltais** [comment]
> I used your settings but learning rates are so high ( 1 instead of usual 0000.1) i get just distorted rectangles as outputs I think webui should have own default settings from get go so we dont have to guess what works and theres no centralised place to store all the settings for all types of locon lycoris loha variants Ok i tried it with recommended rate 1e-2 but results are much worse than what i use with normal lora or locon which is rate default 0.0001 ,batch 3 ,epoch 5, repeats 25-26 image count 30-33

This is because you need to use the Prodify optimizer. It will control the LR... so you set it to 1 and let Prodigy adjust ir via a d parameter to the optimizer.

**Comment by 2blackbar** [comment]
I cant use prodigy, it gives me error 1 and wont start training

**Comment by rafstahelin** [comment]
Bernard, where is the graph from, it is beautiful. Is it from wandb?

**Comment by bmaltais** [comment]
No, it is from tensorboard .
